### PR TITLE
DDF-04909 String Literals on the left on comparisons

### PR DESCRIPTION
**Pull request by dvmarcilio**
#### What does this PR do?
This PR aims to fix issues from SonarQube rule - [Strings literals should be placed on the left side when checking for equality](https://rules.sonarsource.com/java/RSPEC-1132), as first described in #4909.

Transforming from `someString.equals("string")`  to `"string".equals(someString)`. This avoids checking for `null` and also prevents `NullPointerException` being thrown.

I found violations in 45 files, across multiple modules:
- broker (1 file)
- catalog (23 files)
- libs (1 file)
- platform (20 files)

I can gladly break the PR per module, or by any other way that makes review easier.

#### Ask 2 committers to review/merge the PR and tag them here.
Sorry, I don't know who I should tag.

#### Any background context you want to provide?
I asked whether PRs tackling SonarQube issues were welcome on [google groups](https://groups.google.com/forum/?fromgroups#!topic/ddf-developers/Ovdj2lohGow).
#### What are the relevant tickets?
For Jira:
[](https://codice.atlassian.net/browse/)

For GH Issues:
Fixes: #4909 (partially)

#### Checklist:
- [ ] Documentation Updated
- [ ] Update / Add Threat Dragon models
- [ ] Update / Add Unit Tests
- [ ] Update / Add Integration Tests

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.

### CODE REVIEW

File: catalog/core/catalog-core-localstorageprovider/src/main/java/org/codice/ddf/catalog/content/impl/FileSystemStorageProvider.java [hunk 1]
```diff
@@ -467,7 +467,7 @@ private InputStream getInputStreamFromReference(Path externalReferencePath) thro
     Path newPath = null;
     if (reference.getScheme() == null) {
       newPath = Paths.get(reference.toASCIIString());
-    } else if (reference.getScheme().equalsIgnoreCase("file")) {
+    } else if ("file".equalsIgnoreCase(reference.getScheme())) {
```
Comment (coyotesqrl) [inline catalog/core/catalog-core-localstorageprovider/src/main/java/org/codice/ddf/catalog/content/impl/FileSystemStorageProvider.java]: ✏️ This is unnecessary, as the test on line 468 has already guaranteed that `reference.getScheme()` is non null. Sonar should absolutely _not_ have tagged this, regardless of the configuration applied to it.

File: catalog/core/catalog-core-solr/src/main/java/ddf/catalog/source/solr/DynamicSchemaResolver.java [hunk 2]
```diff
@@ -724,7 +724,7 @@ private void addToFieldsCache(Set<AttributeDescriptor> descriptors) {
 
       fieldsCache.add(ad.getName() + schemaFields.getFieldSuffix(format));
 
-      if (!getSpecialIndexSuffix(format).equals("")) {
+      if (!"".equals(getSpecialIndexSuffix(format))) {
```
Comment (coyotesqrl) [inline catalog/core/catalog-core-solr/src/main/java/ddf/catalog/source/solr/DynamicSchemaResolver.java]: ✏️ This is unnecessary; a null format would result in an NPE on line 694 in the stack chain of the call to `getSpecialIndexSuffix`.

File: catalog/ftp/src/main/java/ddf/catalog/ftp/UserManagerImpl.java [hunk 3]
```diff
@@ -197,7 +197,7 @@ private void setAdmin(User user) {
     if (StringUtils.isEmpty(this.admin)) {
       Subject subject = ((FtpUser) user).getSubject();
       for (String role : karafLocalRoles.split(",")) {
-        if (role.equalsIgnoreCase("admin") && subject.hasRole(role)) {
+        if ("admin".equalsIgnoreCase(role) && subject.hasRole(role)) {
```
Comment (coyotesqrl) [inline catalog/ftp/src/main/java/ddf/catalog/ftp/UserManagerImpl.java]: ✏️ This is unnecessary; the elements of the array that results from calling `String.split` are never `null`.

**Comment by dvmarcilio** [comment]
> It is certainly possible you have resolved a handful of actual issues with this PR, but they are lost amongst the noise. If you would remove all the changes that are purely a style preference, leaving only those that resolve potential NPEs, I would be happy to re-review this.

Thank you for the thorough review. I appreciate it.  I confess that I don't have much knowledge about your domain, just followed SonarQube blindly.

I'll address the issues you pointed out on the review in the following days, and hopefully, this PR can bring benefits.

**Comment by rymach** [comment]
agree with @coyotesqrl here
- these `yoda conditions` are harder to read
- what if the original dev didn't think `null` was a possible value?  why check for it? something like @nullable is a more common approach

**Comment by rzwiefel** [comment]
@dvmarcilio Is there any update to this PR or should we close this for now?

**Comment by dvmarcilio** [comment]
@rzwiefel thanks for the reminder! I'm closing it as this would not be beneficial.

**Pull request closed without merging**